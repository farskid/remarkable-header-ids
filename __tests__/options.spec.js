// @ts-nocheck

const Remarkable = require("remarkable");
const HeaderIdsPlugin = require("../index");
const content = require("../mockData");

/**
 * {
        levels,
        anchorClassName: "this-is-anchor-in-header",
        anchorText: "#",
        headerId: (slug) =>
          `header-that-has-this-anchor-${slug}`,
      }
 */

function parseAllHeaders(content, options = {}) {
  return new Remarkable.Remarkable()
    .use(HeaderIdsPlugin(options))
    .render(content);
}

describe("options", () => {
  test("renders with default options", () => {
    expect(parseAllHeaders(content)).toMatchSnapshot();
  });
  test("renders custom anchorClassName", () => {
    expect(
      parseAllHeaders(content, {
        anchorClassName: "test-anchor-className",
      })
    ).toMatchSnapshot();
  });
  test("renders custom anchorText", () => {
    expect(
      parseAllHeaders(content, {
        anchorText: "test-anchor-text",
      })
    ).toMatchSnapshot();
  });
  test("renders custom headerId", () => {
    expect(
      parseAllHeaders(content, {
        headerId: (slug) =>
          `look-I-have-custom-header-id-with-slug-${slug}`,
      })
    ).toMatchSnapshot();
  });
  test("renders with all custom options", () => {
    expect(
      parseAllHeaders(content, {
        anchorClassName: "test-anchor-className",
        anchorText: "test-anchor-text",
        headerId: (slug) =>
          `look-I-have-custom-header-id-with-slug-${slug}`,
      })
    ).toMatchSnapshot();
  });
});
