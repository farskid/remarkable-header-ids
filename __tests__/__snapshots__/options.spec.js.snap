// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options renders custom anchorClassName 1`] = `
"<h1 id=\\"heading-#h1\\"><a class=\\"test-anchor-className\\" id=\\"h1\\" href=\\"#h1\\">#</a>H1</h1>
<p>Some text for h1</p>
<h2 id=\\"heading-#h2\\"><a class=\\"test-anchor-className\\" id=\\"h2\\" href=\\"#h2\\">#</a>H2</h2>
<p>Some text for h2</p>
<h3 id=\\"heading-#h3\\"><a class=\\"test-anchor-className\\" id=\\"h3\\" href=\\"#h3\\">#</a>H3</h3>
<p>Some text for h3</p>
<h4 id=\\"heading-#h4\\"><a class=\\"test-anchor-className\\" id=\\"h4\\" href=\\"#h4\\">#</a>H4</h4>
<p>Some text for h4</p>
<h5 id=\\"heading-#h5\\"><a class=\\"test-anchor-className\\" id=\\"h5\\" href=\\"#h5\\">#</a>H5</h5>
<p>Some text for h5</p>
<h6 id=\\"heading-#h6\\"><a class=\\"test-anchor-className\\" id=\\"h6\\" href=\\"#h6\\">#</a>H6</h6>
<p>Some text for h6</p>
"
`;

exports[`options renders custom anchorText 1`] = `
"<h1 id=\\"heading-#h1\\"><a class=\\"header-anchor\\" id=\\"h1\\" href=\\"#h1\\">test-anchor-text</a>H1</h1>
<p>Some text for h1</p>
<h2 id=\\"heading-#h2\\"><a class=\\"header-anchor\\" id=\\"h2\\" href=\\"#h2\\">test-anchor-text</a>H2</h2>
<p>Some text for h2</p>
<h3 id=\\"heading-#h3\\"><a class=\\"header-anchor\\" id=\\"h3\\" href=\\"#h3\\">test-anchor-text</a>H3</h3>
<p>Some text for h3</p>
<h4 id=\\"heading-#h4\\"><a class=\\"header-anchor\\" id=\\"h4\\" href=\\"#h4\\">test-anchor-text</a>H4</h4>
<p>Some text for h4</p>
<h5 id=\\"heading-#h5\\"><a class=\\"header-anchor\\" id=\\"h5\\" href=\\"#h5\\">test-anchor-text</a>H5</h5>
<p>Some text for h5</p>
<h6 id=\\"heading-#h6\\"><a class=\\"header-anchor\\" id=\\"h6\\" href=\\"#h6\\">test-anchor-text</a>H6</h6>
<p>Some text for h6</p>
"
`;

exports[`options renders custom headerId 1`] = `
"<h1 id=\\"look-I-have-custom-header-id-with-slug-h1\\"><a class=\\"header-anchor\\" id=\\"h1\\" href=\\"#h1\\">#</a>H1</h1>
<p>Some text for h1</p>
<h2 id=\\"look-I-have-custom-header-id-with-slug-h2\\"><a class=\\"header-anchor\\" id=\\"h2\\" href=\\"#h2\\">#</a>H2</h2>
<p>Some text for h2</p>
<h3 id=\\"look-I-have-custom-header-id-with-slug-h3\\"><a class=\\"header-anchor\\" id=\\"h3\\" href=\\"#h3\\">#</a>H3</h3>
<p>Some text for h3</p>
<h4 id=\\"look-I-have-custom-header-id-with-slug-h4\\"><a class=\\"header-anchor\\" id=\\"h4\\" href=\\"#h4\\">#</a>H4</h4>
<p>Some text for h4</p>
<h5 id=\\"look-I-have-custom-header-id-with-slug-h5\\"><a class=\\"header-anchor\\" id=\\"h5\\" href=\\"#h5\\">#</a>H5</h5>
<p>Some text for h5</p>
<h6 id=\\"look-I-have-custom-header-id-with-slug-h6\\"><a class=\\"header-anchor\\" id=\\"h6\\" href=\\"#h6\\">#</a>H6</h6>
<p>Some text for h6</p>
"
`;

exports[`options renders with all custom options 1`] = `
"<h1 id=\\"look-I-have-custom-header-id-with-slug-h1\\"><a class=\\"test-anchor-className\\" id=\\"h1\\" href=\\"#h1\\">test-anchor-text</a>H1</h1>
<p>Some text for h1</p>
<h2 id=\\"look-I-have-custom-header-id-with-slug-h2\\"><a class=\\"test-anchor-className\\" id=\\"h2\\" href=\\"#h2\\">test-anchor-text</a>H2</h2>
<p>Some text for h2</p>
<h3 id=\\"look-I-have-custom-header-id-with-slug-h3\\"><a class=\\"test-anchor-className\\" id=\\"h3\\" href=\\"#h3\\">test-anchor-text</a>H3</h3>
<p>Some text for h3</p>
<h4 id=\\"look-I-have-custom-header-id-with-slug-h4\\"><a class=\\"test-anchor-className\\" id=\\"h4\\" href=\\"#h4\\">test-anchor-text</a>H4</h4>
<p>Some text for h4</p>
<h5 id=\\"look-I-have-custom-header-id-with-slug-h5\\"><a class=\\"test-anchor-className\\" id=\\"h5\\" href=\\"#h5\\">test-anchor-text</a>H5</h5>
<p>Some text for h5</p>
<h6 id=\\"look-I-have-custom-header-id-with-slug-h6\\"><a class=\\"test-anchor-className\\" id=\\"h6\\" href=\\"#h6\\">test-anchor-text</a>H6</h6>
<p>Some text for h6</p>
"
`;

exports[`options renders with default options 1`] = `
"<h1 id=\\"heading-#h1\\"><a class=\\"header-anchor\\" id=\\"h1\\" href=\\"#h1\\">#</a>H1</h1>
<p>Some text for h1</p>
<h2 id=\\"heading-#h2\\"><a class=\\"header-anchor\\" id=\\"h2\\" href=\\"#h2\\">#</a>H2</h2>
<p>Some text for h2</p>
<h3 id=\\"heading-#h3\\"><a class=\\"header-anchor\\" id=\\"h3\\" href=\\"#h3\\">#</a>H3</h3>
<p>Some text for h3</p>
<h4 id=\\"heading-#h4\\"><a class=\\"header-anchor\\" id=\\"h4\\" href=\\"#h4\\">#</a>H4</h4>
<p>Some text for h4</p>
<h5 id=\\"heading-#h5\\"><a class=\\"header-anchor\\" id=\\"h5\\" href=\\"#h5\\">#</a>H5</h5>
<p>Some text for h5</p>
<h6 id=\\"heading-#h6\\"><a class=\\"header-anchor\\" id=\\"h6\\" href=\\"#h6\\">#</a>H6</h6>
<p>Some text for h6</p>
"
`;
